fem/hyperbolic.hpp

```diff
@@ -61,58 +61,102 @@ class FluxFunction
     *
     * @note One can put assertion in here to detect non-physical solution
     */
    virtual real_t ComputeFlux(const Vector &state, ElementTransformation &Tr,
                               DenseMatrix &flux) const = 0;
    /**
-    * @brief Compute normal flux. Optionally overloaded in the
-    * derived class to avoid creating full dense matrix for flux.
+    * @brief Compute normal flux. Optionally overloaded in a derived class to
+    * avoid creating a full dense matrix for flux.
     *
-    * @param[in] state state at the current integration point
-    * @param[in] normal normal vector, @see CalcOrtho
-    * @param[in] Tr face information
+    * @param[in] state state at the current integration point (num_equations)
+    * @param[in] normal normal vector, see mfem::CalcOrtho() (dim)
+    * @param[in] Tr face transformation
     * @param[out] fluxDotN normal flux from the given element at the current
-    * integration point
+    * integration point (num_equations)
     * @return real_t maximum (normal) characteristic velocity
     */
    virtual real_t ComputeFluxDotN(const Vector &state, const Vector &normal,
                                   FaceElementTransformations &Tr,
                                   Vector &fluxDotN) const;
 
    /**
-    * @brief Compute flux Jacobian. Optionally overloaded in the derived class
+    * @brief Compute average flux for over the given interval of states.
+    * Optionally overloaded in a derived class.
+    *
+    * @param[in] state1 state of the beggining of the interval (num_equations)
+    * @param[in] state2 state of the end of the interval (num_equations)
+    * @param[in] Tr element transformation
+    * @param[out] flux_ average flux from the given element at the current
+    * integration point (num_equations, dim)
+    * @return real_t maximum characteristic speed
+    */
+   virtual real_t ComputeAvgFlux(const Vector &state1, const Vector &state2,
+                                 ElementTransformation &Tr,
+                                 DenseMatrix &flux_) const
+   { MFEM_ABORT("Not Implemented."); }
+
+   /**
+    * @brief Compute average normal flux over the given interval of states.
+    * Optionally overloaded in a derived class.
+    *
+    * @param[in] state1 state of the beggining of the interval (num_equations)
+    * @param[in] state2 state of the end of the interval (num_equations)
+    * @param[in] normal normal vector, see mfem::CalcOrtho() (dim)
+    * @param[in] Tr face transformation
+    * @param[out] fluxDotN average normal flux from the given element at the
+    * current integration point (num_equations)
+    * @return real_t maximum (normal) characteristic velocity
+    */
+   virtual real_t ComputeAvgFluxDotN(const Vector &state1, const Vector &state2,
+                                     const Vector &normal,
+                                     FaceElementTransformations &Tr,
+                                     Vector &fluxDotN) const;
```

Comment:
I feel that how one evaluates the average is less a property of the numerical flux function (i.e., it's more a decision to be made by the calling code via the existing `ComputeFlux` function).  Thoughts? Perhaps this it is better and/or necessary to have this method and I just need some clarification as to why.